assets/src/edit-story/elements/image/edit.js

```diff
@@ -41,25 +48,172 @@ const FadedImg = styled.img`
 	${ ImageWithScale }
 `;
 
-const ActualImg = styled.img`
+const CropImg = styled.img`
 	position: absolute;
 	${ ImageWithScale }
 `;
 
-function ImageEdit( { src, origRatio, width, height, x, y, scale, focalX, focalY, rotationAngle, isFullbleed } ) {
+function ImageEdit( { id, src, origRatio, width, height, x, y, scale, focalX, focalY, rotationAngle, isFullbleed } ) {
+	const [ fullImage, setFullImage ] = useState( null );
+	const [ croppedImage, setCroppedImage ] = useState( null );
+	const [ cropBox, setCropBox ] = useState( null );
+
+	const { actions: { setPropertiesById } } = useStory();
+
 	const elementProps = getBox( { x, y, width, height, rotationAngle, isFullbleed } );
 	const imgProps = getImgProps( elementProps.width, elementProps.height, scale, focalX, focalY, origRatio );
+
+	const resizeMoveableRef = useRef();
+	const panMoveableRef = useRef();
+	const translateRef = useRef( [ 0, 0 ] );
+	const cropRef = useRef( [ 0, 0, 0, 0 ] );
+
+	const updatePan = () => {
+		const [ tx, ty ] = translateRef.current;
+		fullImage.style.transform = `translate(${ tx }px, ${ ty }px)`;
+		croppedImage.style.transform = `translate(${ tx }px, ${ ty }px)`;
+	};
+
+	// Refresh moveables to ensure that the selection rect is always correct.
+	useEffect( () => {
+		if ( resizeMoveableRef.current ) {
+			resizeMoveableRef.current.updateRect();
+		}
+		if ( panMoveableRef.current ) {
+			panMoveableRef.current.updateRect();
+		}
+	} );
+
 	return (
 		<Element { ...elementProps }>
-			<FadedImg src={ src } { ...imgProps } />
-			<ActualBox>
-				<ActualImg src={ src } { ...imgProps } />
-			</ActualBox>
+			<FadedImg ref={ setFullImage } draggable={ false } src={ src } { ...imgProps } />
+			<CropBox ref={ setCropBox }>
+				<CropImg ref={ setCroppedImage } draggable={ false } src={ src } { ...imgProps } />
+			</CropBox>
+
+			{ /* Resizable moveable for cropping */ }
+			{ ! isFullbleed && (
```

Comment:
Shouldn't the logic here consider `rotationAngle`, too?

Reply:
`rotationAngle` seems to work correctly now. Do you have some particular feature in mind?

Reply:
Does the cropping work as expected while the image is rotated? For some reason, I expected that this would need some extra handling.

_Testing_

Reply:
I'm seeing this currently:

![croppibg](https://user-images.githubusercontent.com/3294597/71199280-e541fd00-2295-11ea-8e78-0269437d54ac.gif)

Is it working as expected for you?

Reply:
Oh. I see. It's the snap lines that mess with panning. Let me see what can be done here.

Reply:
@miina I had to disable most of bounds and guidelines for the rotated shapes. The horizontal/vertical implementation that Moveable provides is just not good enough UX. I kept the central guides since they look clear enough. The offset calculations can already readjust if the shape is moved out of bounds. I think this looks ok. PTAL.

Reply:
Definitely better and now it works, however, the experience while rotated feels very different:
![cropping](https://user-images.githubusercontent.com/3294597/71262764-c2b8ee00-2340-11ea-882f-241f1571f3c1.gif)

Especially having the boundaries would be great. Would it be very complicated to make custom calculations for that? E.g. to check that the top left corner of the cropbox would not have higher value `x` and `y` coordinates than the visible element, etc.?

One alternative could be removing the rotation (setting it to 0) for the cropping time only, that would potentially make the element move quite a lot though and be annoying for the user.

Or we could leave it as-is for now and deal with the rotation boundaries later, as part of an improvement.

Thoughts?

Reply:
There's no good option here for now. The only good way to express it would be using the rotated boundary box like the red one here:

![Rotated boundaries](https://user-images.githubusercontent.com/726049/71280699-d0ef1600-2310-11ea-96b8-5f039ee9df41.jpg)

That could only be done using a polygon (or a set of bounding lines). Unfortunately, `Moveable` only allows us to specify bounding region using vertical/horizontal lines. We have to then expand the box to fully fit our rotated rectangle (the blue lines). This ends up working like [this demo](https://drive.google.com/file/d/1NPgl0QjtKW-j2BuhVApxIUhVtoYof2Er/view?usp=sharing). I played around with it, and the UX is just disastrous. As a user it's really confusing why some of these boundaries pop up and when.

Perhaps we should if we could contribute the polygon boundaries to the `Moveable` project.

Reply:
Hmm, maybe that's partly a product question in terms of what to do -- if we should support the rotated cropping "partially" since currently the polygon boundaries aren't supported.

To me it feels like from the product perspective we should (eventually) either have it working correctly and as expected or not support it at all, as opposed to having poor/confusing UX.

Maybe for now, we could leave it as-is and connect with the dev of Moveable on contributing to the library to support polygon boundaries.

There's also the (potentially annoying but simpler) alternative of setting the rotation angle to 0 for the time of cropping only. That might also be poor UX, however...

Thoughts?

cc @pbakaus

Reply:
I'd say either:

1. disable rotation while cropping, or
2. implement perfectly (guides, snap points rotate as well)

I was already in touch with the Moveable guy on Twitter, shouldn't be hard to contribute.

Reply:
Okay, in this case, we could merge it as it is and create a follow-up issue to contribute to Moveable to support polygon boundaries for making this complete.